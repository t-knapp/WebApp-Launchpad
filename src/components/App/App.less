@import "../../../assets/styles/Colors.less";

.app {
  h1 {
    color: @appPrimaryColor;
  }

  h2 {
    color: @appSecondaryColor;
  }
}
