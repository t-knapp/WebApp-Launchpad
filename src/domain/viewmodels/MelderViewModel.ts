import IMelderViewModel from './IMelderViewModel';
import {action, observable} from 'mobx';

export default class MelderViewModel implements IMelderViewModel {

    @observable
    public name: string;

    constructor(name: string) {
        this.name = name;
    }

    @action
    public async setName(state: string): void {
        this.name = state;

        try {
            const file = await openFile();
            const fileContent = await readFile(file);

        } catch (ex) {
            dasdasd
        }




        console.error(fileContent);



        return;
    }
}
