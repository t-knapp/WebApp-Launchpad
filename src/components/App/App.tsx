import { ReactElement} from 'react';
import React from 'react';
import emoji from '../../../assets/images/10-denker-rcm600x0.jpg';
import './App.less';
import IMelderViewModel from '../../domain/viewmodels/IMelderViewModel';

type Props = {
    title: string,
    viewModel: IMelderViewModel
};

export default function App(props: Props): ReactElement {
    return (
        <div className='AppComponent'>
            <h1>{props.title}</h1>
            <h2>{props.viewModel.name}</h2>
            <p>
                Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet. Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet.
            </p>
            <h2>Das Gef√ºge der Katastrophen</h2>
            <p>
                Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet. Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet?
                <img src={emoji} width={18} height={18} alt={'Emoji nachdenklich'} />
            </p>
        </div>
    );
}
