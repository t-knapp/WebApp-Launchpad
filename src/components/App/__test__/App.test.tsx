import TestRenderer, {ReactTestRenderer} from 'react-test-renderer';
import React from 'react';
import App from '../App';
import MelderViewModel from '../../../domain/viewmodels/MelderViewModel';

describe('AppComponent', () => {
    describe('render', () => {
        test('should render default', () => {
            const component: ReactTestRenderer = TestRenderer.create(<App viewModel={new MelderViewModel('')} title={'Hallo Welt.'}/>);
            expect(component.toJSON()).toMatchSnapshot();
        });

        test('should render warning state', () => {
            // TODO:
        });

        test('', () => {

        });
    });

    describe('interactions', () => {
        test('should call xyz when click on abc', async (done) => {
            done();
        });
    });
});
