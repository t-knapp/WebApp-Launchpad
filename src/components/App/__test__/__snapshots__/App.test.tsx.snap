// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppComponent render should render default 1`] = `
<div
  className="AppComponent"
>
  <div>
    <p>
      Unser Menue
    </p>
    <p>
      NotToggled
    </p>
    <button
      onClick={[Function]}
    >
      Toggle
    </button>
    <input
      onChange={[Function]}
      type="text"
    />
    <p>
      Value: 
      
    </p>
    <p>
      ViewModel Name: 
      
    </p>
  </div>
  <h1>
    Hallo Welt.
  </h1>
  <h2>
    
  </h2>
  <p>
    Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet. Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet.
  </p>
  <h2>
    Das Gef√ºge der Katastrophen
  </h2>
  <p>
    Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet. Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet?
    <img
      alt="Emoji nachdenklich"
      height={18}
      src="test-file-stub"
      width={18}
    />
  </p>
</div>
`;
